packages/react/src/components/DetailsList/DetailsList.base.tsx

```diff
@@ -845,6 +846,55 @@ export class DetailsListBase extends React.Component<IDetailsListProps, IDetails
     return 0;
   }
 
+  public updateColumn(column: IColumn, newProps: { width?: number; newColumnIndex?: number }) {
```

Comment:
Since the things in `newProps` aren't actually component props, it might be better to just call it `options`.